docs/data/base/components/modal/modal.md

```diff
@@ -0,0 +1,125 @@
+---
+product: base
+title: React Modal component
+components: ModalUnstyled
+githubLabel: 'component: modal'
+waiAria: https://www.w3.org/TR/wai-aria-practices/#dialog_modal
+---
+
+# Modal
+
+<p class="description">The `Modal` component lets you create dialogs, popovers, lightboxes, and other elements that force the user to take action before continuing.</p>
+
+The `Modal` component renders its `children` node in front of a backdrop component. This lets you create an element that your users must interact with before continuing in the parent application.
+Features:
+
+- 💄 Manages modal stacking when more than one is needed
+- 🔐 Creates a backdrop to disable interaction with the rest of the app
+- 🔐 Disables page scrolling while open
+- ♿️ Manages focus correctly between the modal and its parent app
+- ♿️ Adds the appropriate ARIA roles automatically
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+> **Note:** the term "modal" is sometimes used interchangeably with "dialog," but this is incorrect.
+> A dialog may be _modal_ or _nonmodal_.
+> 
+> A modal [blocks interaction with the rest of the application](https://en.wikipedia.org/wiki/Modal_window), forcing the user to take action.
+> As such, it should be used sparingly—only when the app _requires_ user input before it can continue.
+
+If you are creating a modal dialog, the [`Dialog`](/material-ui/dialog/) component is better suited for this specific use case.
+
+`Modal` is a lower-level construct that is leveraged by the following components:
+
+- [Dialog](/material-ui/react-dialog/)
+- [Drawer](/material-ui/react-drawer/)
+- [Menu](/material-ui/react-menu/)
+- [Popover](/material-ui/react-popover/)
+
+## Basic modal
+
+```js
+import ModalUnstyled from '@mui/base/ModalUnstyled';
+```
+
+{{"demo": "ModalUnstyled.js", "defaultCodeOpen": false }}
+
+## Nested modal
+
+> ⚠ **Note:** though it is possible to create nested modals—for example, a select modal within a dialog—stacking more than two at a time is discouraged.
+
+{{"demo": "NestedModal.js", "defaultCodeOpen": false}}
+
+## Transitions
+
+You can animate the open and close states of a modal with a transition component, as long as that component fulfills the following requirements:
+
+- Is a direct child descendent of the modal
+- Has an `in` prop—this corresponds to the open/close state
+- Calls the `onEnter` callback prop when the enter transition starts
+- Calls the `onExited` callback prop when the exit transition is completed
+> The `onEnter` and `onExited` callbacks tell the modal to unmount the child content when closed and fully transitioned.
+
+`Modal` has built-in support for [react-transition-group](https://github.com/reactjs/react-transition-group):
+
+{{"demo": "TransitionsModal.js", "defaultCodeOpen": false}}
+
+Alternatively, you can use [react-spring](https://github.com/pmndrs/react-spring):
```

Comment:
```suggestion
You can also use [react-spring](https://github.com/pmndrs/react-spring) with the `Modal`, but it will require additional custom configuration:
```